esphome/components/http_request/http_request.cpp

```diff
@@ -20,5 +20,25 @@ void HttpRequestComponent::dump_config() {
   }
 }
 
+std::string HttpContainer::get_response_header(const std::string &header_name) {
+  auto response_headers = this->get_response_headers();
+  auto header_name_lower_case = str_lower_case(header_name);
+  if (response_headers.count(header_name_lower_case) == 0) {
+    ESP_LOGD(TAG, "No header with name %s found", header_name_lower_case.c_str());
```

Comment:
If the code explicitly requests a header, and this is not available; shouldn't this be considered at least a warning? Debug logs might be easy to oversee.

Of course you have done more checks; maybe this *is* a frequent case which then would be OK as debug.
```suggestion
    ESP_LOGW(TAG, "No header with name %s found", header_name_lower_case.c_str());
```